packages/ra-ui-materialui/src/list/filter/FilterForm.tsx

```diff
@@ -1,105 +1,41 @@
+    const { filters: filtersProps, ...rest } = props;
     const filters = useContext(FilterContext) || filtersProps;
 
-    const mergedInitialValuesWithDefaultValues =
-        mergeInitialValuesWithDefaultValues(
-            defaultValues || filterValues,
-            filters
-        );
-
-    const form = useForm({
-        defaultValues: mergedInitialValuesWithDefaultValues,
-    });
-    const { getValues, reset, trigger, watch } = form;
-
-    // Reapply filterValues when the URL changes or a user removes a filter
-    useEffect(() => {
-        const newValues = getFilterFormValues(getValues(), filterValues);
-        const previousValues = getValues();
-        if (!isEqual(newValues, previousValues)) {
-            reset(newValues);
-        }
-        // The reference to the filterValues object is not updated when it changes,
-        // so we must stringify it to compare it by value and also compare the reference.
-        // This makes it work for both input values and filters applied directly through
-        // the ListContext.setFilter (e.g. QuickFilter in the simple example)
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [JSON.stringify(filterValues), filterValues, getValues, reset]);
-
-    useEffect(() => {
-        const subscription = watch(async (values, { name }) => {
-            // We must check whether the form is valid as watch will not check that for us.
-            // We can't rely on form state as it might not be synchronized yet
-            const isFormValid = await trigger();
-
-            // Check that the name is present to avoid setting filters when watch was
-            // triggered by a change on the ListContext values.
-            if (name && isFormValid) {
-                if (get(values, name) === '') {
-                    const newValues = cloneDeep(values);
-                    unset(newValues, name);
-                    setFilters(newValues, displayedFilters, true);
-                } else {
-                    setFilters(values, displayedFilters, true);
-                }
-            }
-        });
-        return () => subscription.unsubscribe();
-    }, [displayedFilters, setFilters, trigger, watch]);
-
     return (
-        <FormProvider {...form}>
-            <FormGroupsProvider>
-                <FilterFormBase
-                    onSubmit={handleFormSubmit}
-                    filters={filters}
-                    {...rest}
-                />
-            </FormGroupsProvider>
-        </FormProvider>
+        <AutoSubmitFilterForm
+            component={StyledForm}
+            {...sanitizeRestProps(rest)}
+        >
+            <FilterFormBase filters={filters} />
+        </AutoSubmitFilterForm>
     );
 };
 
-export type FilterFormProps = FilterFormBaseProps & {
-    defaultValues?: FieldValues;
-};
+export type FilterFormProps = FilterFormBaseProps;
 
 export const FilterFormBase = (props: FilterFormBaseProps) => {
```

Comment:
I don't think it makes sense to separate the logic into two components anymore. I suggest you reintegrate FilterFormBase into FilterForm.

Reply:
Actually we still need it because `FilterFormBase` needs to access the form context (`const form = useFormContext()`) created by `AutoSubmitFilterForm`.